### PR TITLE
feat: Add Endpoint for Handling Logged-In User's Password Reset Requests

### PR DESCRIPTION
**Pull request by alexindevs**
#### Summary

This PR introduces a new endpoint to handle password reset requests for logged-in users. The endpoint allows users to change their passwords securely by providing their current password and a new password.

#### Changes

1. **New Endpoint: `POST /api/v1/auth/change-password`**
   - Added a new route in the authentication module to handle password reset requests.
   - The endpoint requires the current password (`oldPassword`) and a new password (`newPassword`) as inputs.

2. **Service Method: `changePassword`**
   - Implemented the `changePassword` method in the `AuthenticationService` to validate and update the user's password.
   - The method includes:
     - Validation of the current password using bcrypt.
     - Hashing of the new password before saving.
     - Exception handling to manage invalid current passwords and other unexpected errors.

4. **Unit Tests**
   - Added unit tests for the `changePassword` method to ensure it handles various scenarios correctly.
   - Test cases include:
     - Successful password change.
     - Handling of invalid current password.
     - Graceful handling of unexpected errors.

#### Affected Files

- `src/modules/auth/auth.controller.ts` (New endpoint)
- `src/modules/auth/auth.service.ts` (Service method)
- `src/modules/auth/tests/auth.service.spec.ts` (Unit tests)

#### Verification Steps

1. **Unit Tests**
   - Run `npm run test` to ensure all test cases pass.
   - Verify that new test cases for the `changePassword` method are correctly implemented.

2. **Manual Testing**
   - Authenticate as a user and make a `POST` request to `/api/v1/auth/change-password` with the current password and new password.
   - Verify that the password is successfully updated when the correct current password is provided.
   - Ensure appropriate error messages are returned for invalid current passwords and other errors.

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -132,6 +132,57 @@ export default class AuthenticationService {
...
+      });
+
+      if (!user) {
+        throw new NotFoundException({
+          status_code: HttpStatus.NOT_FOUND,
+          message: USER_NOT_FOUND,
+        });
+      }
+
+      console.log(
```
Comment (Cobidex) [inline src/modules/auth/auth.service.ts:149]: Remove console.log if it's served it's purpose

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -132,6 +132,57 @@ export default class AuthenticationService {
...
+        oldPassword,
+        'newPassword: ',
+        newPassword,
+        'user.password: ',
+        user.password,
+        bcrypt.compareSync(oldPassword, user.password)
+      );
+
+      const isPasswordValid = bcrypt.compareSync(oldPassword, user.password);
+      console.log('isPasswordValid: ', isPasswordValid);
```
Comment (Cobidex) [inline src/modules/auth/auth.service.ts:160]: Remove

File: src/modules/auth/dto/change-password.dto.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// change-password.dto.ts
```
Comment (Cobidex) [inline src/modules/auth/dto/change-password.dto.ts:1]: Remove dangling comment

File: src/modules/auth/dto/change-password.dto.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+export class ChangePasswordDto {
+  @IsNotEmpty()
+  @IsString()
+  @MinLength(8)
+  oldPassword: string;
+
+  @IsNotEmpty()
+  @IsString()
+  @MinLength(8)
+  newPassword: string;
```
Comment (Cobidex) [inline src/modules/auth/dto/change-password.dto.ts:13]: Add password strength validation rules... Reference the auth register endpoint for clarification.

File: src/modules/auth/tests/auth.service.spec.ts [hunk 5]
```diff
@@ -298,6 +304,80 @@ describe('AuthenticationService', () => {
...
+        first_name: 'John',
+        last_name: 'Doe',
+      };
+    });
+
+    it('should change password successfully', async () => {
+      userServiceMock.getUserRecord.mockResolvedValueOnce(mockUser as User);
+      userServiceMock.updateUserRecord.mockResolvedValueOnce(undefined);
+
+      console.log(oldPassword, newPassword, bcrypt.compareSync(oldPassword, mockUser.password));
```
Comment (Cobidex) [inline src/modules/auth/tests/auth.service.spec.ts:328]: Take off

**Review by PrudentBird — APPROVED** [review]

**Review by rugue — APPROVED** [review]

**Pull request merged**